docs/book/src/reference/submodule-layouts.md

```diff
@@ -0,0 +1,237 @@
+declared in your controller are not taken over into the indirect imports.
+
+### Using replace directives for development
+
+When trying to resolve your main module in the root folder of the operator, you may notice an error if you use a VCS path:
+
+```shell
+go mod tidy
+go: finding module for package YOUR_GO_PATH/test-operator/api/v1alpha1
+YOUR_GO_PATH/test-operator imports
+	YOUR_GO_PATH/test-operator/api/v1alpha1: cannot find module providing package YOUR_GO_PATH/test-operator/api/v1alpha1: module YOUR_GO_PATH/test-operator/api/v1alpha1: git ls-remote -q origin in LOCALVCSPATH: exit status 128:
+	remote: Repository not found.	
+	fatal: repository 'https://YOUR_GO_PATH/test-operator/' not found
+```
+
+The reason for this is that you may have not pushed your modules into the VCS yet and resolving the main module will fail as it can no longer
+directly access the API types as package but only as module.
+
+To solve this issue, we will have to tell the go tooling to properly `replace` the API module with a local reference to your path.
+
+You can do this with 2 different approaches: go modules and go workspaces.
+
+#### Using go modules
+
+For go modules, you will edit the main `go.mod` file of your project and issue a replace directive.
+
+You can do this by editing the go.mod with
+
+```shell
+go mod edit -require YOUR_GO_PATH/test-operator/api/v1alpha1@v0.0.0 # Only if you didn't already resolve the module
+go mod edit -replace YOUR_GO_PATH/test-operator/api/v1alpha1@v0.0.0=./api/v1alpha1
+go mod tidy
+```
+
+Note that we used the placeholder version `v0.0.0` of the API Module. In case you already released your API module once,
+you can use the real version as well. However this will only work if the API Module is already available in the VCS.
+
+<aside class="note warning">
+<h1>Implications on controller releases</h1>
+
+Since the main `go.mod` file now has a replace directive, it is important to drop it again before releasing your controller module.
+To achieve this you can simply run
+
+```shell
+go mod edit -dropreplace YOUR_GO_PATH/test-operator/api/v1alpha1
+go mod tidy
+```
+
+</aside>
+
+#### Using go workspaces
+
+For go workspaces, you will not edit the `go.mod` files yourself, but rely on the workspace support in go.
+
+To initialize a workspace for your project, run ´go work init`.
+
+Now let us include both modules in our workspace:
+```shell
+go work use . # This includes the main module with the controller
+go work use api/v1alpha1 # This is the API submodule
+go work sync
+```
+
+This will lead to commands such as `go run` or `go build` to respect the workspace and make sure that local resolution is used.
+
+You will be able to work with this locally without having to build your module.
+
+When releasing, make sure to set the environment variable `GOWORK=off` (verifiable with `go env GOWORK`) to make sure
+the release process does not get impeded by a potentially commited `go.work` file
```

Comment:
Should `go.work` files ever be committed to a repo? 

If not, maybe it would be best to recommend adding the `go.work` file to the `.gitignore` so there are no accidental commits including it.

Reply:
You are right, i will add a comment here.